Source/diablo.cpp

```diff
@@ -1,1775 +1,1774 @@
-	MSG msg;
-
-	nthread_ignore_mutex(TRUE);
-	start_game(uMsg);
-	/// ASSERT: assert(ghMainWnd);
-	saveProc = SetWindowProc(GM_Game);
-	control_update_life_mana();
-	msg_process_net_packets();
-	gbRunGame = TRUE;
-	gbProcessPlayers = TRUE;
-	gbRunGameResult = TRUE;
-	force_redraw = 255;
-	DrawAndBlit();
-	PaletteFadeIn(8);
-	force_redraw = 255;
-	gbGameLoopStartup = TRUE;
-	nthread_ignore_mutex(FALSE);
-
-	while (gbRunGame) {
-		diablo_color_cyc_logic();
-		if (PeekMessage(&msg, NULL, 0, 0, PM_NOREMOVE)) {
-			SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_ABOVE_NORMAL);
-			while (PeekMessage(&msg, NULL, 0, 0, PM_REMOVE)) {
-				if (msg.message == WM_QUIT) {
-					gbRunGameResult = FALSE;
-					gbRunGame = FALSE;
-					break;
-				}
-				TranslateMessage(&msg);
-				DispatchMessage(&msg);
-			}
-			bLoop = gbRunGame && nthread_has_500ms_passed(FALSE);
-			SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_NORMAL);
-			if (!bLoop) {
-				continue;
-			}
-		} else if (!nthread_has_500ms_passed(FALSE)) {
-			ProcessInput();
-			DrawAndBlit();
-			continue;
-		}
-		multi_process_network_packets();
-		game_loop(gbGameLoopStartup);
-		gbGameLoopStartup = FALSE;
-		DrawAndBlit();
-	}
-
-	if (gbMaxPlayers > 1) {
-		pfile_write_hero();
-	}
-
-	pfile_flush_W();
-	PaletteFadeOut(8);
-	SetCursor_(CURSOR_NONE);
-	ClearScreenBuffer();
-	force_redraw = 255;
-	scrollrt_draw_game_screen(TRUE);
-	saveProc = SetWindowProc(saveProc);
-	/// ASSERT: assert(saveProc == GM_Game);
-	free_game();
-
-	if (cineflag) {
-		cineflag = FALSE;
-		DoEnding();
-	}
-}
-
-void start_game(unsigned int uMsg)
-{
-	zoomflag = TRUE;
-	cineflag = FALSE;
-	InitCursor();
-	InitLightTable();
-	LoadDebugGFX();
-	/// ASSERT: assert(ghMainWnd);
-	music_stop();
-	ShowProgress(uMsg);
-	gmenu_init_menu();
-	InitLevelCursor();
-	sgnTimeoutCurs = 0;
-	sgbMouseDown = 0;
-	track_repeat_walk(0);
-}
-
-void free_game()
-{
-	int i;
-
-	FreeControlPan();
-	FreeInvGFX();
-	FreeGMenu();
-	FreeQuestText();
-	FreeStoreMem();
-
-	for (i = 0; i < MAX_PLRS; i++)
-		FreePlayerGFX(i);
-
-	FreeItemGFX();
-	FreeCursor();
-	FreeLightTable();
-	FreeDebugGFX();
-	FreeGameMem();
-}
-
-void diablo_init(LPSTR lpCmdLine)
-{
-	init_create_window();
-
-	SFileEnableDirectAccess(TRUE);
-	init_archives();
-
-	UiInitialize();
-#ifdef SPAWN
-	UiSetSpawned(TRUE);
-#endif
-
-	ReadOnlyTest();
-
-	srand(GetTickCount());
-	InitHash();
-
-	diablo_init_screen();
-	diablo_parse_flags(lpCmdLine);
-
-	snd_init(NULL);
-	sound_init();
-}
-
-void diablo_splash()
-{
-#ifdef _DEBUG
-	if (!showintrodebug)
-		return;
-#endif
-
-	play_movie("gendata\\logo.smk", TRUE);
-#ifndef SPAWN
-	if (getIniBool("Diablo", "Intro", true)) {
-		play_movie("gendata\\diablo1.smk", TRUE);
-		setIniValue("Diablo", "Intro", "0");
-	}
-#endif
-	UiTitleDialog();
-}
-
-int APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow)
-{
-	diablo_init(lpCmdLine);
-	diablo_splash();
-	mainmenu_loop();
-	UiDestroy();

```

Comment:
You removed this, but are not calling it in your quit function, was that intentional?

Reply:
Look at dvl::init_cleanup().
UiDestroy() called by dvl::init_cleanup() and read request description)